plugins/in_kafka/in_kafka.c

```diff
@@ -50,111 +51,136 @@ static int try_json(mpack_writer_t *writer, rd_kafka_message_t *rkm)
         }
         return ret;
     }
-    mpack_write_object_bytes(writer, buf, bufsize);
+    flb_log_event_encoder_append_body_binary_body(log_encoder, buf, bufsize);
     flb_free(buf);
     return 0;
 }
 
-static void process_message(mpack_writer_t *writer,
-                            rd_kafka_message_t *rkm)
+static int process_message(struct flb_log_event_encoder *log_encoder,
+                           rd_kafka_message_t *rkm)
 {
-    struct flb_time t;
+    int ret;
 
-    mpack_write_tag(writer, mpack_tag_array(2));
+    ret = flb_log_event_encoder_begin_record(log_encoder);
 
-    flb_time_get(&t);
-    flb_time_append_to_mpack(writer, &t, 0);
+    ret = FLB_EVENT_ENCODER_SUCCESS;
+    if (ret == FLB_EVENT_ENCODER_SUCCESS) {
+        ret = flb_log_event_encoder_set_current_timestamp(log_encoder);
+    }
 
-    mpack_write_tag(writer, mpack_tag_map(6));
+    if (ret == FLB_EVENT_ENCODER_SUCCESS) {
+        ret = flb_log_event_encoder_append_body_cstring(log_encoder, "topic");
+    }
 
-    mpack_write_cstr(writer, "topic");
-    if (rkm->rkt) {
-        mpack_write_cstr(writer, rd_kafka_topic_name(rkm->rkt));
+    if (ret == FLB_EVENT_ENCODER_SUCCESS) {
+        if (rkm->rkt) {
+            ret = flb_log_event_encoder_append_body_cstring(log_encoder, rd_kafka_topic_name(rkm->rkt));
+        }
+        else {
+            ret = flb_log_event_encoder_append_body_null(log_encoder);
+        }
     }
-    else {
-        mpack_write_nil(writer);
+
+    if (ret == FLB_EVENT_ENCODER_SUCCESS) {
+        ret = flb_log_event_encoder_append_body_values(log_encoder,
+                                                       FLB_LOG_EVENT_CSTRING_VALUE("partition"),
+                                                       FLB_LOG_EVENT_INT32_VALUE(&rkm->partition));
     }
 
-    mpack_write_cstr(writer, "partition");
-    mpack_write_i32(writer, rkm->partition);
+    if (ret == FLB_EVENT_ENCODER_SUCCESS) {
+        ret = flb_log_event_encoder_append_body_values(log_encoder,
+                                                       FLB_LOG_EVENT_CSTRING_VALUE("offset"),
+                                                       FLB_LOG_EVENT_INT64_VALUE(&rkm->offset));
```

Comment:
`FLB_LOG_EVENT_INT64_VALUE` doesn't expect the value to be a pointer to the value, remove the `&`